### PR TITLE
Added support for specifying a port via code

### PR DESCRIPTION
**Pull request by shnhrrsn**

### CODE REVIEW

**Comment by LoganWright** [comment]
Yes, awesome!

Right now, in Vapor, if someone sets `--ip=`, we automatically use that as the port, do you think we should support that here, or is it assumed that if the user wants to implement curassow, they should use it's implementation via `--bind 127.0.0.1:80

File: README.md [hunk 1]
```diff
@@ -38,11 +38,3 @@ let app = Application()
 app.providers.append(VaporCurassowServer.Provider)
 app.start()
 ```
-
-## Run
-
-Curassow requires some command line arguments to configure ports differently than 8000, here's an example call updating that.
-
-```
-swift build && .build/debug/YourAppName --bind 0.0.0.0:8080
```
Comment (LoganWright) [inline README.md]: Per the other comment, I think I'm leaning towards letting curassow handle the control overrides, so maybe we should leave the `--bind 0.0.0.0:8080` flag for reference with a new comment.

**Comment by shnhrrsn** [comment]
I think we should support it, but I think we should formalize it in Vapor for the server driver + app.start and then implement here

File: Sources/VaporCurassow.swift [hunk 2]
```diff
@@ -123,10 +123,12 @@ public final class CurassowServer: ServerDriver {
     public var delegate: ServerDriverDelegate? = nil
     
     public func boot(port port: Int) throws {
-        if port != 80 {
-            print("Currassow does not support port argument in code, use `--bind 0.0.0.0:\(port)`")
-        }
-        Curassow.serve(handle)
+        var configuration = Configuration()
+        configuration.addresses = [ Address.IP(hostname: "0.0.0.0", port: UInt16(port)) ]
```
Comment (LoganWright) [inline Sources/VaporCurassow.swift]: The alternative would be to stick w/ `--ip=` and `--port=` that're already suppored within Vapor, and override them here.

**Comment by LoganWright** [comment]
Few comments to think about and possibly discuss, then :shipit: when ready

File: Sources/VaporCurassow.swift (hunk 2 above)
Comment (shnhrrsn) [inline Sources/VaporCurassow.swift]: We're not going to be able to support both Curassow CLI options + specifying via code.  Curassow.swift is mostly a CLI wrapper around Arbiter, so by dropping that we lost those features.

Comment (LoganWright) [inline Sources/VaporCurassow.swift]: Cool, then just supporting the current vapor args should be good. Awesome addition!

Do you know anything that might help Curassow halt?

Comment (shnhrrsn) [inline Sources/VaporCurassow.swift]: The methods are here: https://github.com/kylef/Curassow/blob/master/Sources/Arbiter.swift#L89, unfortunately not public.  I'll try submitting a PR in the morning to publicize them.

Comment (shnhrrsn) [inline Sources/VaporCurassow.swift]: @LoganWright Unfortunately calling both `Arbiter.stop()` and `Arbiter.halt()` actually quits the entire process, which isn't what we want.

Looking through Vapor, currently the only thing that makes use of `ServerDriver.halt()` is within `SocketServer.boot()`, so I don't think it's an immediate need.

Comment (LoganWright) [inline Sources/VaporCurassow.swift]: I think it's fine to ship this PR w/o it, I think you can merge whenever you're ready.

**Comment by LoganWright** [comment]
I think this is ready to ship @shnhrrsn 

We should be able to swap

```Swift
configuration.addresses = [ Address.IP(hostname: "0.0.0.0", port: UInt16(port)) ]
```

For 

```Swift
let ip = Process.valueFor("ip") ?? "0.0.0.0"
let p = Process.valueFor("port").flatMap { Int($0) } ?? port
configuration.addresses = [ Address.IP(hostname: ip, port: UInt16(p)) ]
```

**Comment by shnhrrsn** [comment]
Going to submit a PR to clean this up in vapor core then we'll get this merged in

**Comment by shnhrrsn** [comment]
Ready to go now -- I don't have write access, so can't merge on my own.

**Pull request merged**